### PR TITLE
[Retiarii] add validation in base trainers

### PR DESCRIPTION
**Pull request by hzhua**

### CODE REVIEW

File: nni/retiarii/trainer/pytorch/base.py [hunk 1]
```diff
@@ -86,15 +87,23 @@ def __init__(self, model,
...
+                                                             **(dataset_kwargs or {}))
+        self._val_dataset = getattr(datasets, dataset_cls)(train=False,
+                                                           transform=get_default_transform(
+                                                               dataset_cls),
+                                                           **(dataset_kwargs or {}))
         self._optimizer = getattr(torch.optim, optimizer_cls)(
             model.parameters(), **(optimizer_kwargs or {}))
         self._trainer_kwargs = trainer_kwargs or {'max_epochs': 10}
 
         # TODO: we will need at least two (maybe three) data loaders in future.
```
Comment (ultmaster) [inline nni/retiarii/trainer/pytorch/base.py:102]: Remove TODO

File: nni/retiarii/trainer/pytorch/base.py [hunk 2]
```diff
@@ -188,57 +213,64 @@ def __init__(self, multi_model, kwargs=[]):
...
                 return
 
-    
     def training_step(self, batch: Tuple[torch.Tensor, torch.Tensor], batch_idx: int) -> Dict[str, Any]:
         x, y = self.training_step_before_model(batch, batch_idx)
         y_hat = self.model(x)
         return self.training_step_after_model(x, y, y_hat)
 
-    def training_step_before_model(self, batch: Tuple[torch.Tensor, torch.Tensor], batch_idx: int, device = None):
+    def training_step_before_model(self, batch: Tuple[torch.Tensor, torch.Tensor], batch_idx: int, device=None):
```
Comment (ultmaster) [inline nni/retiarii/trainer/pytorch/base.py:273]: Suggest using `self.device`

File: nni/retiarii/trainer/pytorch/base.py [hunk 3]
```diff
@@ -188,57 +213,64 @@ def __init__(self, multi_model, kwargs=[]):
...
-                report_loss[self.kwargs['model_kwargs'][output_idx]['model_id']] = loss.item()
+                report_loss[self.kwargs['model_kwargs']
+                            [output_idx]['model_id']] = loss.item()
             summed_loss = sum(losses)
             summed_loss.backward()
             for opt in self._optimizers:
                 opt.step()
-            if batch_idx % 50 == 0:
-                nni.report_intermediate_result(report_loss)
+            # if batch_idx % 50 == 0:
```
Comment (ultmaster) [inline nni/retiarii/trainer/pytorch/base.py:263]: Why comment this?

**Comment by ultmaster** [comment]
NNI's line limit is 140. You might need to configure your autopep to avoid unwanted linebreaks. :)

File: nni/retiarii/trainer/pytorch/base.py (hunk 1 above)
Comment (hzhua) [reply nni/retiarii/trainer/pytorch/base.py]: removed

File: nni/retiarii/trainer/pytorch/base.py (hunk 2 above)
Comment (hzhua) [reply nni/retiarii/trainer/pytorch/base.py:249]: In MultiModel, different model's input may need to be placed on different devices (called in _train). Currently, the trainer just sets one GPU per model in hard-code.

BTW, train_step and validation_step are not used in PyTorchImageClassificationTrainer. Removed.

File: nni/retiarii/trainer/pytorch/base.py (hunk 3 above)
Comment (hzhua) [reply nni/retiarii/trainer/pytorch/base.py]: It was for debug. training_loss is not reported. Removed.

**Pull request merged**